cluster-autoscaler/core/scaledown/actuation/actuator_test.go

```diff
@@ -708,239 +830,275 @@ func TestStartDeletion(t *testing.T) {
 				"empty-but-with-pods-node-1": {ResultType: status.NodeDeleteErrorInternal, Err: cmpopts.AnyError},
 			},
 		},
-	} {
-		t.Run(tn, func(t *testing.T) {
-			// This is needed because the tested code starts goroutines that can technically live longer than the execution
-			// of a single test case, and the goroutines eventually access tc in fakeClient hooks below.
-			tc := tc
-			// Insert all nodes into a map to support live node updates and GETs.
-			nodesByName := make(map[string]*apiv1.Node)
-			nodesLock := sync.Mutex{}
-			for _, node := range tc.emptyNodes {
-				nodesByName[node.Name] = node
-			}
-			for _, node := range tc.drainNodes {
-				nodesByName[node.Name] = node
-			}
+	}
 
-			// Set up a fake k8s client to hook and verify certain actions.
-			fakeClient := &fake.Clientset{}
-			type nodeTaints struct {
-				nodeName string
-				taints   []apiv1.Taint
-			}
-			taintUpdates := make(chan nodeTaints, 10)
-			deletedNodes := make(chan string, 10)
-			deletedPods := make(chan string, 10)
+	testCasesWithNGNames := map[string]startDeletionTestCase{}
+	for k, v := range testCases {
+		testCasesWithNGNames[k+" "+testNg.Id()] = v
+	}
+
+	return testCasesWithNGNames
+}
+
+func TestStartDeletion(t *testing.T) {
+	testNg1 := testprovider.NewTestNodeGroup("test-ng-1", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	opts1 := &config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: false,
+	}
+	testNg1.SetOptions(opts1)
+	testNg2 := testprovider.NewTestNodeGroup("test-ng-2", 100, 0, 3, true, false, "n1-standard-2", nil, nil)
+	opts2 := &config.NodeGroupAutoscalingOptions{
+		IgnoreDaemonSetsUtilization: true,
+	}
+	testNg2.SetOptions(opts2)
+
+	testSets := []testSet{
+		{
+			nodeGroup: testNg1,
+			testCases: getStartDeletionTestCases(testNg1, opts1.IgnoreDaemonSetsUtilization),
+		},
+		{
+			nodeGroup: testNg2,
+			testCases: getStartDeletionTestCases(testNg2, opts2.IgnoreDaemonSetsUtilization),
+		},
+	}
 
-			ds := generateDaemonSet()
+	for _, testSet := range testSets {
+		for tn, tc := range testSet.testCases {
```

Comment:
I have added a comment for the code I have changed below this line (in this file). The changes are small. Most of the code below this line is same as before, just shifted by some lines.